lib/std/strbasics.nim

```diff
@@ -27,7 +27,7 @@ proc add*(x: var string, y: openArray[char]) =
     i.inc
   # xxx use `nimCopyMem(x[n].addr, y[0].addr, y.len)` after some refactoring
 
-func stripSlice(s: openArray[char], leading = true, trailing = true, chars: set[char] = whitespaces): Slice[int] =
+func stripSlice(s: openArray[char], leading = true, trailing = true, chars: set[char] = whitespaces): Slice[int] {.noinit.} =
```

Comment:
1 question though: couldn't nim compiler (or even C backend) optimize the initialization away instead of having to annotate with `{.noinit.}` ? I'd assume a ton of procs would otherwise need to be annotated with `{.noinit.}` 

in other workds, why single this case out, was there a benchmark justifying it?